lab_3/main.py

```diff
@@ -20,7 +20,38 @@ def tokenize_by_sentence(text: str) -> tuple:
          (('_', 'h', 'e', '_'), ('_', 'i', 's', '_'), ('_', 'h', 'a', 'p', 'p', 'y', '_'))
          )
     """
-    pass
+    if not isinstance(text, str):
+        return ()
+    invaluable_trash = ['`', '~', '@', '#', '$', '%', '^', '&', '*', '(', ')', '-',
+                        '+', '=', '{', '[', ']', '}', '|',
+                        '\\', ':', ';', '"', "'", '<', ',', '>',
+                        '/', '1', '2', '3', '4', '5', '6', '7', '8', '9', '0']
+    text = text.lower()
+    special_letters = {'ü': 'ue', 'ö': 'oe', 'ä': 'ae', 'ß': 'ss'}
+    for letter in text:
```

Comment:
Maybe it would be better to iterate over the special symbols instead? A text can be large.